src/server/db/firebase.js

```diff
@@ -64,14 +64,17 @@ export class Firebase {
       return;
     }
 
-    this.cache.set(id, state);
+    // Remove any undefined values (ie, credentials)
```

Comment:
Are you sure that you're seeing `credentials` here?  That shouldn't even be a part of the state.

It's more likely that you're seeing `undefined` values for `deltalog` and `log`.  These are set in `src/master/master.js`, and can probably just be trimmed away there:

```js
function removeUndefinedFields(state) {
  const { log, deltalog, ...rest } = state;
  return rest;
}
```

Reply:
You're right about `deltalog`. The original error I was seeing was for `undefined` credentials in `deltalog`.

I can update to your suggested fix. Is there anywhere else this may be an issue? Also, it will still be possible for someone to accidentally include an `undefined` and receive the same error but I guess at that point they can address it in their own codebase.

Reply:
I think that one location should do it.